src/components/node-list/index.js

```diff
@@ -83,15 +102,14 @@ class NodeList extends React.Component {
   }
 
   render() {
-    const {
-      onToggleAllNodes,
-      onToggleNodeHovered,
-      onToggleNodeDisabled,
-      nodes,
-      theme
-    } = this.props;
+    const { hasData, nodes, theme } = this.props;
     const { searchValue } = this.state;
-    const formattedNodes = this.highlightMatch(this.filterResults(nodes));
+    if (!hasData) {
+      return null;
+    }
```

Comment:
Should this case be considered here i.e. could this be pushed further up the chain, or does / could this component be used outside the context of kedro viz

Reply:
I added it here intentionally, to prevent things from breaking when no data is provided, or when it's still loading. But it seems to be running fine now when I remove it? I'll doublecheck but it looks like it may no longer be necessary

Reply:
🙂